bliss/predict.py

```diff
@@ -60,34 +60,54 @@ def align(img, sdss):
 def predict_sdss(cfg):
     sdss = instantiate(cfg.predict.dataset)
 
-    sdss_frame = PhotoFullCatalog.from_file(
+    sdss_cat = PhotoFullCatalog.from_file(
         cfg.paths.sdss,
         run=cfg.predict.dataset.run,
         camcol=cfg.predict.dataset.camcol,
         field=cfg.predict.dataset.fields[0],
         sdss_obj=sdss,
     )
 
-    sdss_plocs = sdss_frame.plocs[0]
-    img = sdss[0]["image"]
-    bg = sdss[0]["background"]
-    img = align(img, sdss)
-    bg = align(bg, sdss)
+    sdss_plocs = sdss_cat.plocs[0]
+    sdss[0]["image"] = align(sdss[0]["image"], ref_wcs=sdss[0]["wcs"])
+    sdss[0]["background"] = align(sdss[0]["background"], ref_wcs=sdss[0]["wcs"])
+
+    # mean of the nelec_per_mgy per band
+    nelec_per_nmgy_per_band = np.mean(sdss[0]["nelec_per_nmgy_list"], axis=1)  # ugriz
 
     encoder = instantiate(cfg.encoder).to(cfg.predict.device)
     enc_state_dict = torch.load(cfg.predict.weight_save_path)
     encoder.load_state_dict(enc_state_dict)
     encoder.eval()
     trainer = instantiate(cfg.predict.trainer)
     est_cat, images, background, pred = trainer.predict(encoder, datamodule=sdss)[0].values()
+
+    est_cat = nelec_to_nmgy(est_cat, nelec_per_nmgy_per_band)
     est_full = est_cat.to_full_params()
     if cfg.predict.plot.show_plot and (sdss_plocs is not None):
         ptc = cfg.encoder.tiles_to_crop * cfg.encoder.tile_slen
         cropped_image = images[0, 0, ptc:-ptc, ptc:-ptc]
         cropped_background = background[0, 0, ptc:-ptc, ptc:-ptc]
         plot_predict(cfg, cropped_image, cropped_background, sdss_plocs, est_full)
 
-    return est_cat, images[0], background[0], sdss_plocs, pred
+    return est_full, images[0], background[0], sdss_plocs, pred
+
+
+def nelec_to_nmgy(est_cat, nelec_per_nmgy_per_band):
```

Comment:
Can we name this something more descriptive, like `convert_catalog_from_nelec_to_nmgy`?

Reply:
Done.